securedrop_client/api_jobs/uploads.py

```diff
@@ -28,28 +29,65 @@ def call_api(self, api_client: API, session: Session) -> str:
         we can return the reply uuid.
         '''
         try:
-            encrypted_reply = self.gpg.encrypt_to_source(self.source_uuid, self.message)
-            sdk_reply = self._make_call(encrypted_reply, api_client)
+            draft_reply_db_object = session.query(DraftReply).filter_by(uuid=self.reply_uuid).one()
             source = session.query(Source).filter_by(uuid=self.source_uuid).one()
+            session.commit()
+
+            encrypted_reply = self.gpg.encrypt_to_source(self.source_uuid, self.message)
+            interaction_count = source.interaction_count + 1
+            filename = '{}-{}-reply.gpg'.format(interaction_count,
+                                                source.journalist_designation)
             reply_db_object = Reply(
                 uuid=self.reply_uuid,
                 source_id=source.id,
+                filename=filename,
                 journalist_id=api_client.token_journalist_uuid,
-                filename=sdk_reply.filename,
                 content=self.message,
                 is_downloaded=True,
-                is_decrypted=True
+                is_decrypted=True,
             )
+            sdk_reply = self._make_call(encrypted_reply, api_client)
+
+            # Update filename and file_counter in case they changed on the server
+            new_file_counter = int(sdk_reply.filename.split('-')[0])
+            reply_db_object.file_counter = new_file_counter
+            reply_db_object.filename = sdk_reply.filename
+
+            draft_file_counter = draft_reply_db_object.file_counter
+            draft_timestamp = draft_reply_db_object.timestamp
+
+            update_draft_replies(session, source.id, draft_timestamp,
+                                 draft_file_counter, new_file_counter)
+
+            # Delete draft, add reply to replies table.
             session.add(reply_db_object)
+            session.delete(draft_reply_db_object)
             session.commit()
+
             return reply_db_object.uuid
         except RequestTimeoutError as e:
```

Comment:
This is out of the scope of this PR but shouldn't we also be catching `AuthError` and `ApiInaccessibleError` and raising a custom exception to include reply_uuid and message like we do `SendReplyJobTimeoutError`?

Reply:
:o yes... yes we should